bin/cheribsdtest/cheribsdtest_fptr_canon.c

```diff
@@ -82,3 +84,20 @@ CHERIBSDTEST(fptr_canon_dlfunc,
 
 	cheribsdtest_success();
 }
+
+CHERIBSDTEST(fptr_canon_scalar,
+    "Check that function pointers are canonical when cast to scalar")
```

Comment:
As with LLD, please don't use the term scalar. A capability is still a scalar (contrast with a vector of capabilities).

Reply:
Also, this isn't about casting, this is about loading from memory where the value in memory is an integer address that's been relocated, rather than a pointer.